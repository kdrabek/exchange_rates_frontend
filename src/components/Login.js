import React from 'react';
import { Link } from 'react-router';

import { Column, Colors, Link as ButtonLink, Row } from 'react-foundation';

class Login extends React.Component {
  render() {
    return (
      <Row>
        <Column large={6} offsetOnLarge={3}>
          <h4 class="subheader">Zaloguj się:</h4>

          <form>
            <label>
              Email:
              <input type="email" placeholder="email" />
            </label>
            <label>
              Hasło:
              <input type="password" placeholder="hasło" />
            </label>

            <ButtonLink color={Colors.PRIMARY} isExpanded>Zaloguj się</ButtonLink>

            <Link to="register">
              <a>Nie masz konta? Zarejestruj się.</a>
            </Link>
          </form>
        </Column>
      </Row>
    );
  }
}

export default Login;
