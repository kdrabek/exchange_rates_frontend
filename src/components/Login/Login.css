.errors {
    color: firebrick;
    font-size: 1.2em;
    padding: 10px 0 10px 0;
}