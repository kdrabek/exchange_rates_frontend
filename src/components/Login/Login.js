import React, { Component } from 'react';
import { Link } from 'react-router';
import { Row, Col, FormControl, Button, ControlLabel, FormGroup } from 'react-bootstrap';
import { LocalForm, Control } from 'react-redux-form';
import * as ratesActions from '../../actions/actions';
import { connect } from 'react-redux';

import './Login.css';
import { plLoginErrors } from '../../utils/plLocale';

class Login extends Component {

  handleSubmit(user){
    this.props.dispatch(ratesActions.loginUser(user));
  }

  render() {
    let error = this.props.error ? this.props.error : null;
    return (
      <Row>
        <Col md={6} mdOffset={3}>
          <h4>Logowanie</h4>
          
            <LocalForm onSubmit={(values) => this.handleSubmit(values)}>
              <FormGroup>
                <ControlLabel>Email:</ControlLabel>
                <Control type="email" model=".email" component={FormControl} label="ddd"/>
              </FormGroup>

              <FormGroup>
                <ControlLabel>Hasło:</ControlLabel>
                <Control type="password" model=".password" component={FormControl}/>
              </FormGroup>

              <FormGroup>
                <Button bsStyle="primary" type="submit" block>Zaloguj</Button>
              </FormGroup>
            </LocalForm>

            <div className={ error ? 'errors': ''}>{plLoginErrors[error]}</div>
          
          <Link to="/register">Nie masz konta? Zarejestruj się.</Link>
        </Col>
      </Row>
    );
  }
}

function mapStateToProps(state, ownProps) {
  return {
    error: state.user.error
  };
}

export default connect(mapStateToProps)(Login);
