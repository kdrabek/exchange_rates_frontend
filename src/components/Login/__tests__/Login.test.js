import React from 'react';
import renderer from 'react-test-renderer';
import { reduxForm } from 'redux-form';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { Login } from '../Login';


describe('Login component', () => {

  it('matches the snapshot', () => {
    const DecoratedLogin = reduxForm({ form: 'testForm' })(Login);
    const store = createStore(() => ({}));

    const tree = renderer.create(
      <Provider store={store}>
        <DecoratedLogin loginUser={jest.fn()} />
      </Provider>
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

});