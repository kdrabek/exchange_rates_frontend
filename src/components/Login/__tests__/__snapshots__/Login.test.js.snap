exports[`Login component matches the snapshot 1`] = `
<div
  className="row">
  <div
    className="col-md-6 col-md-offset-3">
    <h4>
      Logowanie
    </h4>
    <form
      className=""
      onSubmit={[Function]}>
      <div
        className="form-group">
        <label
          className="control-label"
          htmlFor={undefined}>
          Email
        </label>
        <input
          className="form-control"
          id={undefined}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div
        className="form-group">
        <label
          className="control-label"
          htmlFor={undefined}>
          Hasło
        </label>
        <input
          className="form-control"
          id={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Hasło"
          type="password"
          value="" />
      </div>
      <button
        className="btn btn-primary btn-block"
        disabled={false}
        type="submit">
        Zaloguj się
      </button>
      <button
        className="btn btn-warning btn-block"
        disabled={true}
        onClick={[Function]}
        type="button">
        Reset
      </button>
    </form>
    <a
      onClick={[Function]}
      style={Object {}}>
      Nie masz konta? Zarejestruj się.
    </a>
  </div>
</div>
`;
